### PR TITLE
Remove LLVM headers in SVF core libs (Partially done) [Stage 1]

### PR DESCRIPTION
**Pull request by whst**
* Remove `llvm::Function` usage in `SVFUtil.h`
* Make project gets compiled without LLVM headers in `SVFUtil.h`
* Move LLVM-dependent functions in `CppUtil.h` to `LLVMUtil.h`

### CODE REVIEW

File: include/SVFIR/SVFType.h [hunk 1]
```diff
@@ -31,6 +31,8 @@
 #ifndef INCLUDE_UTIL_SVFBASICTYPES_H_
 #define INCLUDE_UTIL_SVFBASICTYPES_H_
 
+#include <llvm/Support/Debug.h> // TODO: Remove LLVM Header
```
Comment (yuleisui) [inline include/SVFIR/SVFType.h:34]: Can you remove this header and move to cpp?

File: include/SVF-LLVM/LLVMUtil.h [hunk 2]
```diff
@@ -179,10 +176,12 @@ inline bool ArgInDeadFunction (const Value*  val)
...
 
 /// Return true if this is an argument of a program entry function (e.g. main)
-inline bool ArgInProgEntryFunction (const Value*  val)
+inline bool ArgInProgEntryFunction(const Value* val)
 {
-    return SVFUtil::isa<Argument>(val)
-           && SVFUtil::isProgEntryFunction(SVFUtil::cast<Argument>(val)->getParent());
+    return SVFUtil::isa<Argument>(val) &&
+           SVFUtil::isProgEntryFunction(
+               LLVMModuleSet::getLLVMModuleSet()->getSVFFunction(
```
Comment (yuleisui) [inline include/SVF-LLVM/LLVMUtil.h:183]: Do not use llvm module here, define a isProgEntryFunction in llvm util

File: include/SVFIR/SVFType.h (hunk 1 above)
Comment (whst) [reply include/SVFIR/SVFType.h]: Removed.

File: include/SVF-LLVM/LLVMUtil.h (hunk 2 above)
Comment (whst) [reply include/SVF-LLVM/LLVMUtil.h]: Added another `llvm::Function` version in `LLVMUtil.h`.

File: lib/SVF-LLVM/LLVMUtil.cpp [hunk 3]
```diff
@@ -139,7 +149,8 @@ bool LLVMUtil::isUncalledFunction (const Function*  fun)
 {
     if(fun->hasAddressTaken())
         return false;
-    if(SVFUtil::isProgEntryFunction(fun))
+    if (SVFUtil::isProgEntryFunction(
```
Comment (yuleisui) [inline lib/SVF-LLVM/LLVMUtil.cpp:152]: Use LLVMUtil::isProgEntryFunction

**Pull request merged**